columnq/src/columnq.rs

```diff
@@ -5,13 +5,47 @@ use std::sync::Arc;
 use datafusion::arrow;
 use datafusion::arrow::array::as_string_array;
 use datafusion::arrow::array::StringArray;
+use datafusion::datasource::object_store::{ObjectStoreRegistry, ObjectStoreProvider};
+// use datafusion::error::DataFusionError;
 pub use datafusion::execution::context::SessionConfig;
 use datafusion::execution::context::SessionContext;
+use datafusion::execution::runtime_env::{RuntimeConfig, RuntimeEnv};
 use datafusion::physical_plan::collect;
 
+use object_store::aws::AmazonS3Builder;
 use crate::error::{ColumnQError, QueryError};
 use crate::query;
 use crate::table::{self, KeyValueSource, TableSource};
+use url::Url;
+
+pub struct ColumnQObjectStoreProvider {}
+impl ObjectStoreProvider for ColumnQObjectStoreProvider {
+    fn get_by_url(&self, url: &Url) -> Option<Arc<dyn object_store::ObjectStore>> {
+        let url_schema = url.scheme();
+        match url_schema {
+            "s3" => {
+                let host = url.host_str().unwrap();
+                // TODO: can remove "with_endpoint" after object_store upgrade to 0.5.3
```

Comment:
note: currently, `from_env` supports `AWS_ENDPOINT` only. But after 0.5.3, it supports both `AWS_ENDPOINT` or `AWS_ENDPOINT_URL`